File: packages/capacitor-notificare-assets/ios/Plugin/NotificareAssetsPlugin.swift
import Foundation
import Capacitor
import NotificareKit
import NotificareAssetsKit

@objc(NotificareAssetsPlugin)
public class NotificareAssetsPlugin: CAPPlugin {
    @objc func fetch(_ call: CAPPluginCall) {
        guard let group = call.getString("group") else {
            call.reject("Missing 'group' parameter.")
            return
        }
        
        Notificare.shared.assets().fetch(group: group) { result in
            switch result {
            case let .success(assets):
                do {
                    call.resolve([
                        "result": try assets.map { try $0.toJson() }
                    ])
                } catch {
                    call.reject(error.localizedDescription)
                }
            case let .failure(error):
                call.reject(error.localizedDescription)
            }
        }
    }
}

File: packages/capacitor-notificare-assets/android/src/main/java/re/notifica/assets/capacitor/NotificareAssetsPlugin.kt
package re.notifica.assets.capacitor

import com.getcapacitor.*
import com.getcapacitor.annotation.CapacitorPlugin
import re.notifica.Notificare
import re.notifica.NotificareCallback
import re.notifica.assets.ktx.assets
import re.notifica.assets.models.NotificareAsset

@CapacitorPlugin(name = "NotificareAssetsPlugin")
public class NotificareAssetsPlugin : Plugin() {
    @PluginMethod
    public fun fetch(call: PluginCall) {
        val group = call.getString("group") ?: run {
            call.reject("Missing 'group' parameter.")
            return
        }

        Notificare.assets().fetch(group, object : NotificareCallback<List<NotificareAsset>> {
            override fun onSuccess(result: List<NotificareAsset>) {
                try {
                    call.resolve(
                        JSObject().apply {
                            put("result", JSArray().apply {
                                result.forEach { put(it.toJson()) }
                            })
                        }
                    )
                } catch (e: Exception) {
                    call.reject(e.localizedMessage)
                }
            }

            override fun onFailure(e: Exception) {
                call.reject(e.localizedMessage)
            }
        })
    }
}
